### PR TITLE
docs(backend): Reflect latest changes in authorization for SQL API

### PR DESCRIPTION
**Pull request by ovr**
Hello!

Thanks

### CODE REVIEW

File: docs/content/Configuration/Config.mdx [hunk 1]
```diff
@@ -267,6 +268,34 @@ module.exports = {
 };
 ```
 
+### <--{"id" : "Options Reference"}--> checkAuth
+
+Used in [SQL API][ref-sql-api]. Can be an `async` function. Default implementation verify username & password from environment variables: `CUBE_SQL_USERNAME`, `CUBE_SQL_PASSWORD`,
```
Comment (rpaik) [inline docs/content/Configuration/Config.mdx:273]:
```suggestion
Used in [SQL API][ref-sql-api], and can be an `async` function. Default implementation verify username & password from environment variables: `CUBE_SQL_USERNAME`, `CUBE_SQL_PASSWORD`,
```

File: docs/content/Cube.js-Backend/SQL-API.mdx [hunk 2]
```diff
@@ -47,42 +47,48 @@ services:
...
-$ npx cubejs-cli token
-Warning. Option --user-context was deprecated and payload will be stored inside root instead of u
-Generating Cube.js JWT token
-
------------------------------------------------------------------------------------------
-  Use these manually generated tokens in production with caution.
-  Please refer to https://cube.dev/docs/security for production security best practices.
------------------------------------------------------------------------------------------
+<DangerBox>
+Before the v0.28.55 version, Cube SQL used the same JWTs as Cube, but we moved to credentials authentication as a workaround of username
```
Comment (rpaik) [inline docs/content/Cube.js-Backend/SQL-API.mdx:52]:
```suggestion
Prior to the v0.28.55 version, Cube SQL used the same JWTs as Cube, but we moved to credentials authentication as a workaround for username
```

**Comment by rpaik** [comment]
A couple of minor suggestions/nits. 😄

File: docs/content/Configuration/Config.mdx [hunk 3]
```diff
@@ -267,6 +268,34 @@ module.exports = {
 };
 ```
 
+### <--{"id" : "Options Reference"}--> checkAuth
```
Comment (hassankhan) [inline docs/content/Configuration/Config.mdx:271]:
```suggestion
### <--{"id" : "Options Reference"}--> checkSqlAuth
```

File: docs/content/Cube.js-Backend/SQL-API.mdx (hunk 2 above)
Comment (ovr) [reply docs/content/Cube.js-Backend/SQL-API.mdx]: Thanks!

File: docs/content/Configuration/Config.mdx (hunk 3 above)
Comment (ovr) [reply docs/content/Configuration/Config.mdx]: Thanks!

File: docs/content/Configuration/Config.mdx (hunk 1 above)
Comment (ovr) [reply docs/content/Configuration/Config.mdx:273]: Thanks!

**Pull request merged**